wear/src/main/java/io/homeassistant/companion/android/tiles/ThermostatTile.kt

```diff
@@ -0,0 +1,311 @@
+                        thermostatTileDao.add(updated)
+                        targetTemp = updatedTargetTemp
+                    } else {
+                        val updated = tileConfig.copy(targetTemperature = null)
+                        thermostatTileDao.add(updated)
+                    }
+
+                    tile.setTileTimeline(
+                        timeline(
+                            tileConfig,
+                            entity,
+                            targetTemp,
+                            temperatureUnit
+                        )
+                    ).build()
+                } catch (e: Exception) {
+                    Log.e(TAG, "Unable to fetch entity ${tileConfig.entityId}", e)
+
+                    tile.setTileTimeline(
+                        primaryLayoutTimeline(
+                            this@ThermostatTile,
+                            requestParams,
+                            null,
+                            R.string.tile_fetch_entity_error,
+                            R.string.refresh,
+                            ActionBuilders.LoadAction.Builder().build()
+                        )
+                    ).build()
+                }
+            }
+        }
+    }
+
+    override fun onTileResourcesRequest(requestParams: ResourcesRequest): ListenableFuture<Resources> = serviceScope.future {
+        Resources.Builder()
+            .setVersion(requestParams.version)
+            .addIdToImageMapping(
+                RESOURCE_REFRESH,
+                ResourceBuilders.ImageResource.Builder()
+                    .setAndroidResourceByResId(
+                        ResourceBuilders.AndroidImageResourceByResId.Builder()
+                            .setResourceId(io.homeassistant.companion.android.R.drawable.ic_refresh)
+                            .build()
+                    ).build()
+            )
+            .build()
+    }
+
+    override fun onTileAddEvent(requestParams: EventBuilders.TileAddEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            val dao = AppDatabase.getInstance(this@ThermostatTile).thermostatTileDao()
+            if (dao.get(requestParams.tileId) == null) {
+                dao.add(ThermostatTile(id = requestParams.tileId))
+            } // else already existing, don't overwrite existing tile data
+        }
+    }
+
+    override fun onTileRemoveEvent(requestParams: EventBuilders.TileRemoveEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            AppDatabase.getInstance(this@ThermostatTile)
+                .thermostatTileDao()
+                .delete(requestParams.tileId)
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        serviceScope.cancel()
+    }
+
+    private suspend fun timeline(tileConfig: ThermostatTile, entity: Entity<Map<String, Any>>, targetTemperature: Float?, temperatureUnit: String): Timeline =
```

Comment:
No suspending functions used

```suggestion
    private fun timeline(tileConfig: ThermostatTile, entity: Entity<Map<String, Any>>, targetTemperature: Float?, temperatureUnit: String): Timeline =
```